stl/src/syserror_import_lib.cpp

```diff
@@ -51,10 +51,18 @@ _NODISCARD size_t __CLRCALL_PURE_OR_STDCALL __std_system_error_allocate_message(
     if (_Ret == 0) {
         _Lang_id = 0;
     }
-    const unsigned long _Chars =
+    unsigned long _Chars =
         FormatMessageA(FORMAT_MESSAGE_ALLOCATE_BUFFER | FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS,
             nullptr, _Message_id, _Lang_id, reinterpret_cast<char*>(_Ptr_str), 0, nullptr);
-
+    if (_Chars == 0 && _Lang_id != 0) {
+        const DWORD _Last_error = GetLastError();
+        if (_Last_error == ERROR_MUI_FILE_NOT_FOUND || _Last_error == ERROR_RESOURCE_LANG_NOT_FOUND) {
+            LocalFree(*_Ptr_str);
+            _Chars = FormatMessageA(
+                FORMAT_MESSAGE_ALLOCATE_BUFFER | FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS, nullptr,
+                _Message_id, 0, reinterpret_cast<char*>(_Ptr_str), 0, nullptr);

```

Comment:
```suggestion
            const DWORD _US_lang_id = MAKELANGID(LANG_ENGLISH, SUBLANG_ENGLISH_US);
            _Chars                  = FormatMessageA(
                FORMAT_MESSAGE_ALLOCATE_BUFFER | FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS, nullptr,
                _Message_id, _US_lang_id, reinterpret_cast<char*>(_Ptr_str), 0, nullptr);
```

Could you test this? Does it work for you?
It works for me...

Reply:
This fails if the US english locale is not installed (which it isn't on localized windows installs)

Reply:
Did you test it? What windows versions?
I believe this should work because:
1) I tested it on a localized windows install(Russian)
2) Look at https://learn.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-formatmessagea

```
 If you pass in zero, FormatMessage looks for a message for LANGIDs in the following order:

1. Language neutral
2. Thread LANGID, based on the thread's locale value
3. User default LANGID, based on the user's default locale value
4. System default LANGID, based on the system default locale value
5. US English
```
US English is the final fallback. This is why I think it should work.
Of course I may be wrong and more real tests are needed.

Reply:
My program used to hardcode US english as well, but I had to change that because I would get failures on some systems (I tried a brand-new Japanese install of the OS and saw the issue, IIRC): https://github.com/TranslucentTB/TranslucentTB/commit/2d1cd95a7c3ff1062c675ae7aa744722a28a74de

Reply:
Is is possible to add a test to avoid regression?

Reply:
Our test infra (both internal and external) doesn't really make it possible to test system locales other than en-US.